### DISCUSSION TITLE
An experience of farming with pools and my idea about switch between farming pools

### DISCUSSION THREAD

**Discussion by tgarm** [Chia-Network/chia-blockchain#4020] [Ideas]
### The experience
In recent days, I've made a small test with 2 existing farming pools ( **hpool** and **uupool** ). 
In order to optimize my revenue, I signed my key to both pools and installed their closed-source miner application to some of my computers, trying to keep similar farming power to each pool and see which one produces the better payout.

Unfortunately, after a few days, both of my accounts were suspended. And my balance in these pools are frozen, not able to withdraw. One (hpool) announced they suspended the "dual farming" accounts. It seems I was identified as "dual farming". So I lost my coins and wasted my farming power. 

This tragedy  was because both pools are **not** designed well to keep safe from dual-farm (farm the same plots at the same time through different pools or solo). And it seems difficult for any pool to keep away from dual-farm while the plots were kept away from being locked in any pool.
This problem needs to be solved for the increasing network farming power makes solo mining almost impossible. Farmers should be free, and pools should be able to know the actual farming power a farmer provides, no dual.

### My idea

- Switch between farms should be freely, but **not** totally free
- A transaction should be initiated and mined in the Chia blockchain, to indicate plots with a certain pool public key is now joined to one pool
- When the farmer wants to quit and join another pool, another transaction should be  mined to indicate the change

Is this a good idea? Thank you for reading, and please let me know if this is feasible.

**Comment by racabu** [comment]
I would suggest something different. Why not the actual Chia network does not pay all miners a percentage based on proof of time and proof of space at the time a block is mined? Instead of randomly choosing someone. That would completely remove the need for a pool and also be fair with small miner operations. Helping to decentralize much more the network. Currently, the network is rewarding the ones who have the most plots (proportionally). So it will make an individual farmer migrate to a pool and concentrate the network instead of making it descentralized.

**Reply by ilyakooo0** [reply]
I imagine the problem will be identifying that a person actually controls the percentage of space they claim they control. Obviously, this can not be self-reported – everyone would say they control as much as possible. The farmers would need to prove that they control the space that they control for _every_ transaction (payout). This is obviously not feasible – it would have tremendous communication and/or computation complexities. 

And not just that. "Controlling" some potion of space on its own does absolutely nothing for the network. Space needs to actually be used for farming. This is also really hard to prove efficiently.

A solution might be to let probability do the work – make the probability of payouts be proportional to the amount of space controlled. Then, in the long run, everyone would get paid the exact same amount that they would get in your scheme, but this time it is actually possible to do it securely and fairly. This is what we have now.

---

You have to remember – there are no centralized authorities that determine how exactly rewards should be split among farmers.

**Reply by unitedchia** [reply]
Although will we also run a pool([chiaunited.com](chiaunited.com)), I would like to see the user base split equally across all trusted pools to limit centralization while allowing the small ones to make profit.

**Reply by tgarm** [reply]
In my understanding, for a blockchain system, only one reward can be distributed at a single time. So maybe the "random" is already the fairest way to reward farmers till now. 

For a storage based farming system, "dual mining" can be avoided if any plot--pool relationship is known at any block height (solo-mining can be treated as using one's solo pool). 

Currently, this relationship is not known for the blockchain system. If we make it recorded as transactions in the blockchain system, the dual-mining problem can be solved, and farmers can be freely switch among farming pools.

Pool is required, we just want it to be better.
